src/js/view/utils/views-manager.js

```diff
@@ -91,6 +100,18 @@ export default {
     remove: function(view) {
         removeFromGroup(view, views);
     },
+    addScrollHandler: function(handler) {
+        if (!scrollHandlerInitialized) {
+            scrollHandlerInitialized = true;
+            window.addEventListener('scroll', onScroll);
+        }
+        if (scrollHandlers.indexOf(handler) === -1) {
```

Comment:
this is a safeguard for any faulty logic, maybe not needed?

Reply:
I prefer to always remove listeners before adding to avoid duplicates. You could add the `if (scrollHandlers.indexOf(handler) === -1) {` check below in `removeScrollHandler`.